tools/lint.py

```diff
@@ -0,0 +1,157 @@
+    ".*operators/generic/transpose/cutt/calls.h",
+    ".*operators/generic/transpose/cutt/cuttGpuModel.h",
+    ".*operators/generic/transpose/cutt/cuttGpuModel.cc",
+    ".*operators/generic/transpose/cutt/cuttGpuModelKernel.h",
+    ".*operators/generic/transpose/cutt/cuttGpuModelKernel.cu",
+    ".*operators/generic/transpose/cutt/CudaMemcpy.h",
+    ".*operators/generic/transpose/cutt/CudaMemcpy.cu",
+    ".*operators/generic/transpose/cutt/CudaUtils.h",
+    ".*operators/generic/transpose/cutt/CudaUtils.cu",
+    ".*operators/generic/transpose/cutt/cuttTypes.h",
+    ".*operators/generic/transpose/cutt/int_vector.h",
+    ".*operators/generic/transpose/cutt/LRUCache.h",
+    ".*python/dummy.cu"
+]
+
+
+def negative_filtering(patterns: list, file_list):
+    """
+    Patterns shall be a list of regex patterns
+    """
+    if len(patterns) == 0:
+        return file_list
+    prog = re.compile(patterns.pop())
+    it = (i for i in file_list if not prog.search(i))
+    return negative_filtering(patterns, it)
+
+
+def gather_files(path: str, patterns: list, antipatterns: list):
+    """
+    Gather files, based on `path`, that match `patterns` unix-like specification
+    and do not match `antipatterns` regexes
+    """
+    curr_path = os.getcwd()
+    os.chdir(path)
+    positive_iterators = [glob.iglob(os.path.join('**', pattern), recursive=True) for pattern in
+                          patterns]
+    linted_files = itertools.chain(*positive_iterators)
+    linted_files = (os.path.join(path, file) for file in linted_files)
+    linted_files = negative_filtering(antipatterns.copy(), linted_files)
+    ret = list(linted_files)
+    os.chdir(curr_path)
+    return ret
+
+
+def gen_cmd(dali_root_dir, file_list, process_includes=False):
+    """
+    Command for calling cpplint.py
+    """
+    cmd = ["python",
+           os.path.join(dali_root_dir, "third_party", "cpplint.py"),
+           "--quiet",
+           "--linelength=100",
+           "--root=" + os.path.join(dali_root_dir, "include" if process_includes else "")]
+    cmd.extend(file_list)
+    return cmd
+
+
+def lint(dali_root_dir, file_list, process_includes, n_subprocesses):
+    """
+    n_subprocesses: how many subprocesses to use for linter processing
+    Returns: 0 if lint passed, 1 otherwise
+    """
+    subprocesses = []
+    diff = int(len(file_list) / n_subprocesses)
+    for process_idx in range(n_subprocesses - 1):
+        subprocesses.append(subprocess.Popen(
+            gen_cmd(dali_root_dir=dali_root_dir,
+                    file_list=file_list[process_idx * diff: (process_idx + 1) * diff],
+                    process_includes=process_includes)))
+    subprocesses.append(subprocess.Popen(
+        gen_cmd(dali_root_dir=dali_root_dir,
+                file_list=file_list[(n_subprocesses - 1) * diff:],
+                process_includes=process_includes)))
+    ret = 0
+    for sp in subprocesses:
+        ret += sp.wait()
+    return 0 if ret == 0 else 1
```

Comment:
`return ret` ?

Reply:
This function is supposed to return proper error code. `1` is the proper error code for generic error, so I can't return `2` for example. Anyhow, this is actually wrong also, since `1` and `-1` will give good result, thanks for pointing out.